frontend/src/main/scala/bloop/engine/tasks/Result.scala

```diff
@@ -0,0 +1,59 @@
+package bloop.engine.tasks
+
+/** The result of the execution of a `Task`. */
+sealed trait Result[+T] {
```

Comment:
What is the difference between this and `Try[T]`?

Monix Task catches exceptions in map/flatMap, `Task.materialize` can turn Task[T] into `Task[Try[T]]`. It also has combinators to recover from errors.

Reply:
Good find, thanks!